cfme/configure/settings.py

```diff
@@ -53,272 +49,280 @@ def __init__(self, description=None, scope=None, days=None, hours=None, timezone
+                row[0].check()
+        view.timeprofile_form.configuration.item_select("Delete selected "
+                                                        "Time Profiles", handle_alert=True)
 
 
 @navigator.register(Timeprofile, 'All')
 class TimeprofileAll(CFMENavigateStep):
+    VIEW = TimeProfileAddFormView
     prerequisite = NavigateToAttribute('appliance.server', 'MySettings')
 
     def step(self):
-        tabs.select_tab("Time Profiles")
-
-
-@navigator.register(Timeprofile, 'Add')
-class TimeprofileNew(CFMENavigateStep):
-    prerequisite = NavigateToSibling('All')
-
-    def step(self):
-        cfg_btn('Add a new Time Profile')
-
-
-@navigator.register(Timeprofile, 'Edit')
-class TimeprofileEdit(CFMENavigateStep):
-    prerequisite = NavigateToSibling('All')
-
-    def step(self):
-        timeprofile_table.click_cell("description", self.obj.description)
+        self.view.mysetting.tabs.time_profile.select()
 
 
 class Visual(Updateable, Navigatable):
-
-    pretty_attrs = ['name']
-
-    item_form = Form(
-        fields=[
-            ('grid_view', AngularSelect('perpage_grid')),
-            ('tile_view', AngularSelect("perpage_tile")),
-            ('list_view', AngularSelect("perpage_list")),
-            ('reports', AngularSelect("perpage_reports")),
-        ])
-
-    startpage_form = Form(
-        fields=[
-            ('login_page', AngularSelect("start_page"))])
-
-    quadicons_form = Form(
-        fields=[
-            ('infra_provider_quad', CFMECheckbox("quadicons_ems")),
-            ('cloud_provider_quad', CFMECheckbox("quadicons_ems_cloud")),
-            ('host_quad', CFMECheckbox("quadicons_host")),
-            ('datastore_quad', CFMECheckbox("quadicons_storage")),
-            ('datastoreitem_quad', Input("quadicons_storageitem")),
-            ('vm_quad', CFMECheckbox("quadicons_vm")),
-            ('vmitem_quad', Input("quadicons_vmitem")),
-            ('template_quad', CFMECheckbox("quadicons_miq_template")),
-        ])
-
-    display_form = Form(
-        fields=[
-            ('chart_theme', AngularSelect("display_reporttheme")),
-            ('time_zone', AngularSelect("display_timezone")),
-        ])
-
-    save_button = form_buttons.FormButton("Add this Time Profile")
-
     @property
     def grid_view_limit(self):
-        navigate_to(self, 'All')
-        return int(re.findall("\d+", self.item_form.grid_view.first_selected_option_text)[0])
+        view = navigate_to(self, 'All')
+        value = re.findall("\d+", view.visualitem.grid_view.read())
+        return int(value[0])
 
     @grid_view_limit.setter
     def grid_view_limit(self, value):
-        navigate_to(self, 'All')
-        fill(self.item_form.grid_view, str(value))
-        sel.click(form_buttons.save)
+        view = navigate_to(self, 'All')
+        value_to_fill = VersionPick({
+            Version.lowest(): "{} items".format(str(value)),
+            '5.8': str(value)
+        }).pick(self.appliance.version)
```

Comment:
REQUIRED: Use `if self.appliance_version < '5.8': ...` instead of VersionPick, since VersionPick is designed to be used on classes and using it like this looks clumsy.